hal/halx86/acpi/halpnpdd.c

```diff
@@ -243,7 +243,8 @@ HalpQueryDeviceRelations(IN PDEVICE_OBJECT DeviceObject,
                 }
 
                 /* Free existing structure */
-                ExFreePool(*DeviceRelations);
+                // TODO: Would there be a known/unique tag to use? TAG_HAL (See 9e478e9 (r46628))?
+                ExFreePoolWithTag(*DeviceRelations, 0);
```

Comment:
It may depend from where the data was allocated...

Reply:
ExFreePoolWithTag cannot be used there because the original tag is not known.

Reply:
Updated comment for acpi and legacy files.